C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackBotWorker.cs

```diff
@@ -29,7 +51,31 @@ public SlackBotWorker(Botkit botkit, object config) : base(botkit, config)
         /// <param name="userId">A Slack user id, like one found in `message.user` or in a `<@mention>`</param>
         public async Task<object> StartPrivateConversation(string userId)
         {
-            return new object();
+            try
+            {
+                JoinDirectMessageChannelResponse actual = null;
+
+                // TO-DO: await this.api.im.open({ user: userId });
+                this.api.JoinDirectMessageChannel(response =>
+                {
+                    actual = response;
+                }, userId);
+
+                var convRef = new ConversationReference();
```

Comment:
This _conversation reference_ should be provided in the constructor, via the _config object_